packages/react/src/components/List/List.tsx

```diff
@@ -1,24 +1,65 @@
-import type { ComponentPropsWithoutRef } from 'react';
-import React from 'react';
+import type { HTMLAttributes, ReactNode } from 'react';
+import React, { useId } from 'react';
 import cn from 'classnames';
 
+import { Heading, Paragraph } from '../Typography';
+
 import classes from './List.module.css';
 
 export type ListProps = {
-  /** Select which border style between items*/
-  borderStyle?: 'solid' | 'dashed';
-} & ComponentPropsWithoutRef<'ul'>;
+  /**
+   * The type of list to render.
+   * @default ul
+   */
+  as?: 'ul' | 'ol';
+  /** Changes text sizing
+   * @default medium
+   */
+  size?: 'small' | 'medium' | 'large';
+  /**
+   * Heading above the list
+   */
+  heading?: string;
```

Comment:
Maybe this should be the same type as for `Heading` aka. `ReactNode`?

Reply:
So they send the whole heading as a prop?

Reply:
Not necessarily, but users might want to use an icon or add some additional text styling (`abbr`, `strong` etc.) which is why most of the time we use the `ReactNode` type.[ Allowed children for heading elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content)

In some use-cases we want to force or not allow anything else than a `string` because its imperative that it is just a text for example `a11y` reasons, such as `content` in `Tooltip`.

Reply:
Goctha makes sense, updating :)